packages/eth-format/index.ts

```diff
@@ -63,6 +67,23 @@ export function findBestUnit(wei: bigint): EtherUnit {
   return "wei";
 }
 
+/**
+ * This function converts an array of `wei` values to the best unit
+ * and returns the converted values and the best unit.
+ */
+export function arrayBestUnit(arr: number[]): {
+  unit: EtherUnit;
+  converted: string[];
+} {
+  const maxVal = Math.max(...arr);
+  const unit = findBestUnit(maxVal);
+
+  return {
+    converted: arr.map((value) => convertWei(value, unit)),
+    unit,
+  };
+}
+
```

Comment:
No need to have a new function to do this.

I'd  update the `findBestUnit` function to also support array inputs of types bigint, number, or string, and perform the logic defined in lines 78-79 when it receives an array.

Additionally, I would create a `convertWeis` function that accepts an array of bigint, number, or string and applies the logic defined in line 82.

Reply:
> I'd update the findBestUnit function to also support array inputs of types bigint, number, or string, and find the best unit in them
https://github.com/Blobscan/blobscan/commit/1f10cf1310398aded44f1f0cbbe2ef91026788a9#diff-e63748d4ac678f6bf062ade6788391a3bdc455132dd96343eb414bf3cde1b993R20


> Additionally, I would create a convertWeis function that accepts an array of bigint, number, or string and applies the logic defined in line 82.
https://github.com/Blobscan/blobscan/commit/1f10cf1310398aded44f1f0cbbe2ef91026788a9#diff-e63748d4ac678f6bf062ade6788391a3bdc455132dd96343eb414bf3cde1b993R24